yaksh/views.py

```diff
@@ -1229,20 +1229,33 @@ def courses(request):
 @login_required
 def course_detail(request, course_id):
     user = request.user
+    ci = RequestContext(request)
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
     course = get_object_or_404(Course, creator=user, pk=course_id)
-    return my_render_to_response('yaksh/course_detail.html', {'course': course})
+    return my_render_to_response('yaksh/course_detail.html', {'course': course},
+                                context_instance=ci)
 
 
 @login_required
-def enroll(request, course_id, user_id, was_rejected=False):
+def enroll(request, course_id, user_id=None, was_rejected=False):
     user = request.user
+    ci = RequestContext(request)
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
     course = get_object_or_404(Course, creator=user, pk=course_id)
-    user = get_object_or_404(User, pk=user_id)
-    course.enroll(was_rejected, user)
+    if request.method == 'POST':
+        data = request.POST.getlist('check')
+        if data is None:
+            return my_render_to_response('yaksh/course_detail.html', {'course': course},
+                                        context_instance=ci)
+        else:
+            for i in data:
+                user = get_object_or_404(User, pk=i)
```

Comment:
the user object has to be present since we are only passing the ids through the context.
instead of for loop and checking object exists or not better to use id__in as the filter.